metis-common/metis-common-mongo/src/main/test/java/eu/europeana/metis/mongo/utils/MorphiaUtilsTest.java

```diff
@@ -0,0 +1,208 @@
+package eu.europeana.metis.mongo.utils;
+
+import static dev.morphia.query.experimental.filters.Filters.eq;
+import static dev.morphia.query.experimental.filters.Filters.or;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.model.Collation;
+import dev.morphia.Datastore;
+import dev.morphia.Morphia;
+import dev.morphia.aggregation.experimental.Aggregation;
+import dev.morphia.aggregation.experimental.AggregationOptions;
+import dev.morphia.annotations.Entity;
+import dev.morphia.annotations.Id;
+import dev.morphia.query.FindOptions;
+import dev.morphia.query.Query;
+import eu.europeana.metis.mongo.embedded.EmbeddedLocalhostMongo;
+import eu.europeana.metis.mongo.model.HasMongoObjectId;
+import java.util.List;
+import java.util.Objects;
+import org.bson.types.ObjectId;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit test for {@link MorphiaUtils}
+ *
+ * @author Jorge Ortiz
+ * @since 31-01-2022
+ */
+class MorphiaUtilsTest {
+
+  private final static String DATABASE_NAME = "dbTest";
+
+  private static Datastore datastore;
+
+  private static EmbeddedLocalhostMongo embeddedLocalhostMongo;
+
+  @BeforeAll
+  static void setup() {
+    embeddedLocalhostMongo = new EmbeddedLocalhostMongo();
+    embeddedLocalhostMongo.start();
+    final String mongoHost = embeddedLocalhostMongo.getMongoHost();
+    final int mongoPort = embeddedLocalhostMongo.getMongoPort();
+    final MongoClient mongoClient = MongoClients.create(String.format("mongodb://%s:%s", mongoHost, mongoPort));
+    datastore = Morphia.createDatastore(mongoClient, DATABASE_NAME);
+  }
+
+  @AfterAll
+  static void tearDown() {
+    embeddedLocalhostMongo.stop();
+  }
+
+  @BeforeEach
```

Comment:
Do you need to be doing this before each and after each?
Can't this be done once for all tests?

Reply:
Thanks. for pointing it out. it can be simplified. done!